clang/test/CXX/basic/basic.types/p10.cpp

```diff
@@ -142,7 +142,7 @@ constexpr int arb(int n) { // expected-note {{declared here}}
                expected-note {{function parameter 'n' with unknown value cannot be used in a constant expression}}
 }
 constexpr long Overflow[(1 << 30) << 2]{}; // expected-warning {{requires 34 bits to represent}} \
-                                              expected-warning {{variable length array folded to constant array as an extension}} \
+                                              expected-error {{variable length array declaration not allowed at file scope}} \
```

Comment:
Hm. Why do we stop folding here but start folding in enumerator values?

Reply:
It depends on what the caller asks for.  TryToFixInvalidVariablyModifiedType calls EvaluateAsInt directly in a mode that doesn't allow undefined behavior, so it's just treated as an unevaluatable expression.  A few other places use CheckConvertedConstantExpression, which is also strict.  Enumerators go through VerifyIntegerConstantExpression, which is not strict.

Reply:
Perhaps we should file an issue to come back and improve consistency here in the future? The behavior is likely mysterious from a user perspective.

In terms of losing the extension, I think it's unlikely to cause significant disruption for anyone so I don't think it needs to be addressed immediately.

Reply:
The diagnostic shouldn't be hard to fix; it's just a matter of using the form of Evaluate() that returns notes.

For the actual behavior here... we could pass the flag to allow undefined behavior when we do this folding, I guess, but I'd rather not.  Or if we do, it should be a DefaultError diagnostic.